### PR TITLE
A simple block list CLI

### PR DESCRIPTION
**Pull request by hansl**
Initial version

### CODE REVIEW

File: many-block/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+[package]
+name = "many-block"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[dependencies]
+chrono = "0.4.19"
```
Comment (fmorency) [inline many-block/Cargo.toml]: Please update to `>=0.4.20`.

`0.4.19` is impacted by https://rustsec.org/advisories/RUSTSEC-2020-0159

```suggestion
chrono = "0.4.22"
```

File: many-block/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+clap = { version = "3.2.6", features = [ "derive" ] }
+hex = "0.4.3"
+humantime = "2.1.0"
+many = { git = "https://github.com/liftedinit/many-rs.git" }
+many-client = { git = "https://github.com/liftedinit/many-rs.git" }
```
Comment (fmorency) [inline many-block/Cargo.toml]: Please fix a `rev`.

File: many-block/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+minicbor = "0.18.0"
+tabled = "0.7.0"
+tracing = "0.1.35"
+tracing-subscriber = "0.2.24"
```
Comment (fmorency) [inline many-block/Cargo.toml]: Please use `0.3.15`.

```suggestion
tracing-subscriber = "0.3.15"
```

**Review by fmorency — CHANGES_REQUESTED** [review]
Building fails because of the latest `many-rs` merge. I can't test.

File: many-block/Cargo.toml (hunk 2 above)
Comment (hansl) [reply many-block/Cargo.toml]: Done.

File: many-block/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    /// The server to list the blocks from.
+    server: ManyUrl,
+
+    /// Number of blocks to query (default to 30).
+    #[clap(long, default_value = "30")]
+    count: u64,
+
+    /// Max Height. By default, this will be the last block (calling blockchain.info).
+    #[clap(long)]
+    max_height: Option<u64>,
```
Comment (fmorency) [inline many-block/src/main.rs]: Error if `count >= max_height`

File: many-block/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+                "".to_string()
+            };
+            last_block_time = Some(block.timestamp.0);
+
+            BlockRow {
+                height: block.id.height,
+                tx_count: block.txs_count,
+                app_hash: block
+                    .app_hash
+                    .map_or("-".to_string(), |h| hex::encode(&h).to_string()),
```
Comment (fmorency) [inline many-block/src/main.rs]:
```suggestion
                    .map_or("-".to_string(), |h| hex::encode(&h)),
```

**Review by fmorency — APPROVED** [review]
LGTM, assuming comments have been addressed!

- Add `Cargo.lock`

File: many-block/src/main.rs (hunk 4 above)
Comment (hansl) [reply many-block/src/main.rs]: It doesn't error. I just clip it. Clarified.

**Pull request merged**